lib/engine/game/g_18_texas/step/simultaneous_auction.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../step/auctioner'
+
+module Engine
+  module Game
+    module G18Texas
+      module Step
+        class SimultaneousAuction < Engine::Step::Base
+          include Engine::Step::Auctioner
+
+          attr_reader :companies
+
+          def actions(entity)
+            return [] if !entity.player? || (entity != current_entity) || !bids_for_player(entity).empty?
+
+            ['bid']
+          end
+
+          def auctioning
+            nil
+          end
+
+          def description
+            'Bid on Companies'
+          end
+
+          def available
+            @companies
+          end
+
+          def skip!
+            current_entity.pass!
+            next_entity!
+          end
+
+          def next_entity!
+            return all_passed! if entities.all?(&:passed?)
+
+            @round.next_entity_index!
+            entity = entities[entity_index]
+            next_entity! if entity&.passed?
+          end
+
+          def process_bid(action)
+            action.entity.pass!
+
+            company = action.company
+            price = action.price
+            entity = action.entity
+            @log << "#{entity.name} bids #{@game.format_currency(price)} for #{company.name}"
+
+            # unpass previous bidder
+            @bids[company].first&.entity&.unpass!
+            replace_bid(action)
+            next_entity!
+          end
+
+          def auctioneer?
+            false
+          end
+
+          def min_increment
+            @game.class::MIN_BID_INCREMENT
+          end
+
+          def setup
+            setup_auction
+            @companies = @game.companies.dup
+          end
+
+          def starting_bid(company)
+            [0, company.value].max
+          end
+
+          def min_bid(company)
+            return unless company
+
+            return starting_bid(company) unless @bids[company].any?
```

Comment:
if empty

Reply:
Hey, I ripped that off from 1817.

Reply:
fix it in both places and stop copying code :)